app/controllers/submissions_controller.rb

```diff
@@ -28,8 +28,14 @@ def create
     @author_id = params[:author_id]
     if @submission.save
       if @author_id
-        authorship = Authorship.new(person_id: @author_id,submission: @submission)
-        authorship.save
+	 submission = Submission.new(@submission)
+-        submission.save
+	 if @submission.corresponding_author_email
+          SubmissionMailer.confirmation(submission).deliver_now    
+          redirect_to submission
+	 else
+	  render :new
+	 end
```

Comment:
Inna kwestia jest taka, że ten kod powinien być w `public_submissions_controller` ale możemy to pominąć.